### PR TITLE
ref(ui): Consolidate server frontend hydration logic

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This gets us one step closer to being able to boot the application without the backed serving the layout.html

### CODE REVIEW

File: src/sentry/static/sentry/app/bootstrap.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import 'bootstrap/js/alert';
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/bootstrap.jsx]: This is pretty much 100% just a rename from app/index.jsx, it just needed to be a diff module

File: src/sentry/static/sentry/app/index.js [hunk 2]
```diff
@@ -1,153 +1,15 @@
...
-
-// setup jquery for CSRF tokens
-jQuery.ajaxSetup({
-  //jQuery won't allow using the ajaxCsrfSetup function directly
-  beforeSend: ajaxCsrfSetup,
-});
-
-// App setup
+// Initial data hydration. The __initialData will be set by the django backend
+// serving sentry.
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/index.js]: Probably easier to read this if you just look at the new file

File: src/sentry/static/sentry/app/index.js [hunk 3]
```diff
@@ -1,153 +1,15 @@
...
-  OrganizationsLoader: require('app/components/organizations/organizationsLoader')
-    .default,
-};
-
-// Make globals available on the window object
-Object.keys(globals).forEach(name => (window[name] = globals[name]));
-
-export default globals;
+// Once data hydration is done we can initialize the app
+require('app/bootstrap');
```
Comment (BYK) [inline src/sentry/static/sentry/app/index.js]: Wouldn't it be better if we wrapped all the init code in `app/bootstrap` above in a function and called it here instead giving us clearer module imports and more controlled init in tests?

This should also allow us to pass initial data as an argument to that function and allow us to stop relying on the `window` object for init in the fully separate FE scenario.

File: src/sentry/static/sentry/app/stores/configStore.jsx [hunk 4]
```diff
@@ -43,15 +46,13 @@ const ConfigStore = Reflux.createStore({
...
       }
 
-      // Priority:
-      // "?lang=en" --> user configuration options --> django request.LANGUAGE_CODE --> "en"
-      setLocale(
-        queryString.lang || config.user.options.language || languageOverride || 'en'
-      );
-    } else if (languageOverride) {
-      setLocale(languageOverride);
+      languageCode = queryString.lang || config.user.options.language || languageCode;
```
Comment (BYK) [inline src/sentry/static/sentry/app/stores/configStore.jsx]: TODO: see if we can also leverage [`navigator.language`](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language) as a last resort.

**Review by BYK — APPROVED** [review]
☠️ what can go wrong?

File: src/sentry/static/sentry/app/index.js (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/index.js]: Unfortunately the window.__* stuff is used in some other modules that are imported within the bootstrap.jsx. So it's a little more complicated, but yes I generally do agree.

File: src/sentry/static/sentry/app/bootstrap.jsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+// SDK INIT  --------------------------------------------------------
+Sentry.init({
+  ...window.__SENTRY__OPTIONS,
+  integrations: [
+    new ExtraErrorData({
+      // 6 is arbitrary, seems like a nice number
+      depth: 6,
+    }),
+    new Tracing({
+      tracingOrigins: ['localhost', 'sentry.io', /^\//],
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/bootstrap.jsx]: 👍 but lets do in a different PR, this was all just copied over.

File: src/sentry/static/sentry/app/bootstrap.jsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+import ajaxCsrfSetup from 'app/utils/ajaxCsrfSetup';
+import plugins from 'app/plugins';
+
+// SDK INIT  --------------------------------------------------------
+Sentry.init({
+  ...window.__SENTRY__OPTIONS,
+  integrations: [
+    new ExtraErrorData({
+      // 6 is arbitrary, seems like a nice number
+      depth: 6,
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/bootstrap.jsx]: 👍 but lets do in a different PR, this was all just copied over.

File: src/sentry/static/sentry/app/bootstrap.jsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+  $: jQuery,
+  jQuery,
+
+  // django templates make use of these globals
+  createReactClass,
+};
+
+// The SentryApp global contains exported app modules for use in javascript
+// modules that are not compiled with the sentry bundle.
+globals.SentryApp = {
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/bootstrap.jsx]: I separated this out a while ago to make things more clear.

**Pull request merged**